src/sentry/testutils/relay.py

```diff
@@ -101,15 +106,15 @@ def post_and_retrieve_event(self, data):
             json=data,
         )
 
-        assert resp.ok
+        if not resp.ok:
```

Comment:
Can you turn this around? Basically have `post_and_retrieve_event` raise errors, and `post_and_try_retrieve_event` catch them down. Otherwise, you will miss invalid status code responses.

Reply:
Not sure I understand your argument, need to check with you.
When the request returns a **not** OK post_and_try... will return `None` and post_and_retrieve_event will fail the assertion.

Reply:
ok, refactored